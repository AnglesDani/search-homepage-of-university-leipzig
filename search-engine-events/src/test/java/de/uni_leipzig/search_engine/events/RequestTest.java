package de.uni_leipzig.search_engine.events;

import java.util.HashMap;
import java.util.Map;

import org.approvaltests.Approvals;
import org.junit.Test;

public class RequestTest
{
	@Test(expected=NullPointerException.class)
	public void checkRequestCouldNotBeCreatedFromNull()
	{
		new Request(null);
	}
	
	@Test
	public void approveRequestCreationWithoutHeaders()
	{
		Approvals.verify(new Request(TestUtils.mockServletRequest("http://www.example.com")));
	}
	
	@Test
	public void approveRequestCreationWithHeaders()
	{
		Map<String, String> headers = new HashMap<>();
		headers.put("a", "value-1");
		headers.put("b", "value-2");
		headers.put("c", "value-3");
		
		Approvals.verify(new Request(TestUtils.mockServletRequest("http://www.example.com", headers)));
	}
}
