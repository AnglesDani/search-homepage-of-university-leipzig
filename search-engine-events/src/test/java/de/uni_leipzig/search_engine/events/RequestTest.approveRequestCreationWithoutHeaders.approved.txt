Request(headers={}, requestUrl=http://www.example.com, remoteAddr=0.0.0.0, remoteHost=localhost, remotePort=1234)